app/assets/javascripts/discourse/app/widgets/post-cooked.js

```diff
@@ -378,38 +376,39 @@ export default class PostCooked {
     return cookedDiv;
   }
 
-  _initUserStatusOnMentions() {
+  _decorateMentions(cooked) {
     if (!this._isInComposerPreview) {
-      this._trackMentionedUsersStatus();
-      this._rerenderUserStatusOnMentions();
+      destroyUserStatusOnMentions();
     }
-  }
 
-  _rerenderUserStatusOnMentions() {
-    destroyUserStatusOnMentions();
-    this._post()?.mentioned_users?.forEach((user) =>
-      this._rerenderUserStatusOnMention(this.cookedDiv, user)
-    );
-  }
+    this._post()?.mentioned_users?.forEach((user) => {
```

Comment:
Most of the refactoring here is done to avoid iterating multiple times on the list of users, and also sharing the mentions nodes